### PR TITLE
Fix module: proteus

### PR DESCRIPTION
**Pull request by WackerO**
While the module itself is not buggy, using the measure_col_prefix param in a pipeline is tricky because nextflow has some whitespace trimming enabled. This PR makes proteus flexible by checking the prefix directly as it is input as well as with an additional whitespace at the end. It also reduces a bit the mem usage by not reading the intensity table fully when it only needs the header.

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/proteus/readproteingroups/templates/proteus_readproteingroups.R [hunk 1]
```diff
@@ -218,16 +221,20 @@ sample.sheet\$condition <- sample.sheet[[opt\$contrast_variable]]
...
 
+if (!any(measure.cols %in% colnames(intensities.table))) {
+    measure.cols <- setNames(paste0(paste0(opt\$measure_col_prefix, " "), sample.sheet[[opt\$sample_id_col]]), sample.sheet[[opt\$sample_id_col]])
+}
+
 # Check that all samples specified in the input sheet are present in the intensities table
 
-missing_columns <- paste0(opt\$measure_col_prefix, sample.sheet[[opt\$sample_id_col]])
-missing_columns <- missing_columns[!missing_columns %in% colnames(intensities.table)]
+#missing_columns <- paste0(opt\$measure_col_prefix, sample.sheet[[opt\$sample_id_col]])
```
Comment (pinin4fjords) [inline modules/nf-core/proteus/readproteingroups/templates/proteus_readproteingroups.R:230]: Just remove the commented line

**Review by pinin4fjords — APPROVED** [review]
Looks fine, approving, but please strip the commented line before merging

File: modules/nf-core/proteus/readproteingroups/templates/proteus_readproteingroups.R (hunk 1 above)
Comment (WackerO) [reply modules/nf-core/proteus/readproteingroups/templates/proteus_readproteingroups.R]: Oops, good catch, forgot about that!

**Comment by WackerO** [comment]
@pinin4fjords Is the conda test absolutely required to work before we can merge this? I'm afraid I won't get it to work, the one solution that I found online did not help at all. Also, docker and singularity ran perfectly fine so far...

**Comment by pinin4fjords** [comment]
> @pinin4fjords Is the conda test absolutely required to work before we can merge this? I'm afraid I won't get it to work, the one solution that I found online did not help at all. Also, docker and singularity ran perfectly fine so far...

Since you mentioned to me that this is reproducible locally, we can infer that letting this slide would disable execution of the differentialabundance workflow under conda when using this module, which I don't think we can tolerate. The plotting components triggering the failure are not crucial to the main output of the workflow, so could be removed if a better solution can't be found.

**Pull request merged**